### PR TITLE
Enforce lowercase name in radixconfig

### PR DESCRIPTION
**Pull request by olemoi**
Applications with uppercase /metadata/name in radixconfig.yaml will trigger a warning in the build process stating that the name contains uppercase letters and automatically convert it to lowercase for  the user.

### CODE REVIEW

File: pkg/apis/radixvalidators/validate_ra_test.go [hunk 1]
```diff
@@ -32,6 +33,43 @@ func Test_missing_rr(t *testing.T) {
...
+
+			if err != nil {
+				assert.False(t, isValid)
+				assert.NotNil(t, err)
+				assert.True(t, testcase.expectedError.Error() == err.Error())
+				assert.True(t, strings.ToLower(validRA.Name) == expectedName)
+			} else {
+				assert.True(t, isValid)
+				assert.Nil(t, err)
+			}
```
Comment (JoakimHagen) [inline pkg/apis/radixvalidators/validate_ra_test.go:68]: maybe also add `assert.True(t, validRA.Name == expectedName)`?

File: pkg/apis/radixvalidators/validate_ra_test.go [hunk 2]
```diff
@@ -32,6 +33,43 @@ func Test_missing_rr(t *testing.T) {
...
+		name          string
+		expectedError error
+		updateRa      updateRAFunc
+	}{
+		{"Mixed case name", radixvalidators.ApplicationNameNotLowercaseError(mixedCaseName), func(ra *v1.RadixApplication) { ra.Name = mixedCaseName }},
+		{"Lower case name", radixvalidators.ApplicationNameNotLowercaseError(lowerCaseName), func(ra *v1.RadixApplication) { ra.Name = lowerCaseName }},
+		{"Upper case name", radixvalidators.ApplicationNameNotLowercaseError(upperCaseName), func(ra *v1.RadixApplication) { ra.Name = upperCaseName }},
+	}
+
+	// _, client := validRASetup()
```
Comment (JoakimHagen) [inline pkg/apis/radixvalidators/validate_ra_test.go:53]: is this old code?

File: pkg/apis/radixvalidators/validate_ra_test.go (hunk 1 above)
Comment (olemoi) [reply pkg/apis/radixvalidators/validate_ra_test.go:68]: The validator does not actively convert the name, it just checks if the name contains uppercase characters. The conversion takes place in `apply_radixconfig.go`

File: pkg/apis/radixvalidators/validate_ra_test.go (hunk 2 above)
Comment (olemoi) [reply pkg/apis/radixvalidators/validate_ra_test.go]: Yes, removed.

File: pkg/apis/radixvalidators/validate_ra_test.go (hunk 1 above)
Comment (JoakimHagen) [reply pkg/apis/radixvalidators/validate_ra_test.go:67]: You're right.

**Pull request merged**